File: meetmind/agents/backend.py
"""Backend Engineer agent."""

from __future__ import annotations

from meetmind.agents.base import BaseAgent
from meetmind.config.constants import BACKEND


class BackendAgent(BaseAgent):
    def __init__(self) -> None:
        super().__init__(name=BACKEND, role="后端工程师")

    @property
    def system_prompt(self) -> str:
        return (
            "你是项目的【后端工程师】。\n"
            "你的核心职责：\n"
            "  1. 设计/评估 API 接口（路径、入参、返回结构、错误码）。\n"
            "  2. 设计数据模型与数据库 schema、索引、迁移策略。\n"
            "  3. 评估性能、并发、安全（鉴权、注入、限流）风险。\n"
            "  4. 给出实现思路与工时预估（粗略）。\n\n"
            "工作风格：\n"
            "  - 用代码片段、ER 图描述、接口表格等具体形式说话，不空谈。\n"
            "  - 充分利用你 RAG 中已有的代码风格和约定，保持一致。\n"
            "  - 主动指出和前端/测试/PM 的协作点。\n\n"
            "结束发言后通常应把发言权交回 architect 汇总。"
        )

File: src/agents/backend.ts
/**
 * 后端工程师 Agent。
 */

import { BACKEND } from "../config/constants.js";
import { BaseAgent } from "./base.js";

export class BackendAgent extends BaseAgent {
  constructor() {
    super(BACKEND, "后端工程师");
  }

  get systemPrompt(): string {
    return (
      "你是项目的后端工程师。核心职责是设计或评估 API 接口（路径、入参、返回结构、错误码），设计数据模型与数据库 schema、索引和迁移策略，评估性能、并发、安全（鉴权、注入、限流）风险，给出粗略的实现思路与工时预估。\n" +
      "工作风格上用代码片段、ER 图描述、接口表格等具体形式说话不空谈，主动指出和前端、测试、PM 的协作点。\n" +
      "只有在真正展开后端方案、确实需要对齐已有代码风格与约定时，才查 RAG 工具；其余情况直接作答，不要为了答一句话去检索。\n" +
      "如果用户只是打招呼、闲聊或问与技术无关的问题，就用一句话自然回应，不要主动堆砌专业知识，也不要调用任何工具。\n" +
      "结束发言后通常把发言权交回 architect 汇总，回复控制在 150 字以内。"
    );
  }
}
